### PR TITLE
Correctly handle ipv6 mapped ipv4 addresses when using recvmmsg

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

394a1b3485000c211595aff7495c4f863972af29 introduced the possibility to use recvmmsg(...) but did not correctly handle ipv6 mapped ip4 addresses to make it consistent with other transports.

Modifications:

- Correctly handle ipv6 mapped ipv4 addresses by only copy over the relevant bytes
- Small improvement on how to detect ipv6 mapped ipv4 addresses by using memcmp and not byte by byte compare
- Adjust test to cover this bug

Result:

Correctly handle ipv6 mapped ipv4 addresses

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please (tests failed due maven error with connection reset :/)

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java [hunk 1]
```diff
@@ -117,6 +117,8 @@ void release() {
         private int count;
 
         private final byte[] addr = new byte[16];
+        private final byte[] ipv4Bytes = new byte[4];
```
Comment (njhill) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: Maybe add a comment to make it clear this is just a temp array to construct ipv4 `InetAddress`? kind of different to the other "value" fields

Comment (normanmaurer) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: yep let me do this...

Comment (normanmaurer) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: let me even make it part of the outer-class to save some memory.

Comment (normanmaurer) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: @njhill done

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java [hunk 2]
```diff
@@ -145,10 +150,11 @@ private void init(long memoryAddress, int count, InetSocketAddress recipient) {
 
         DatagramPacket newDatagramPacket(ByteBuf buffer, InetSocketAddress localAddress) throws UnknownHostException {
             final InetAddress address;
-            if (scopeId != 0) {
-                address = Inet6Address.getByAddress(null, addr, scopeId);
+            if (addrLen == ipv4Bytes.length) {
+                System.arraycopy(addr, 0, ipv4Bytes, 0, addrLen);
```
Comment (johnou) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: Don't we need to sync on ipv4Bytes over these two operations so it's thread safe?

Comment (normanmaurer) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java]: Nope all the access here is single threaded

**Pull request merged**